### PR TITLE
✨🚧 [WIP] implement the abillity to swap solution cards by clicking

### PR DESCRIPTION
**Pull request by robpal94**
**Changes:**

1. Add the abillity to swap solution cards in the solution explorer. 

PR: #PullRequest

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 1]
```diff
@@ -45,6 +45,9 @@ export class SolutionExplorerList {
    * This service is also put inside the map.
    */
   private _singleDiagramService: SingleDiagramsSolutionExplorerService;
+
+  private _clickedSolutionIndices: Array<number> = [];
```
Comment (Paulomart) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]: Its not required to use a array here.  You only need to keep a single number/instance.

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 2]
```diff
@@ -73,7 +73,17 @@ export class SolutionExplorerList {
     (window as any).solutionList = this;
   }
 
-  public moveSelection(solutionEntry: ISolutionEntry): void {
+  public moveSelection(solutionEntry: ISolutionEntry, event: Event): void {
+    const clickedElement: HTMLElement = event.target as HTMLElement;
+    const clickedElementClassList: DOMTokenList = clickedElement.classList;
+    const userClickedOnHeader: boolean =
+      clickedElementClassList.contains('solution-entry__header__name')
+      || clickedElementClassList.contains('solution-entry__header__icon');
```
Comment (SteffenKn) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]: This indentation just looks wrong. Not quite sure if this is better but what about something like:
```
    const userClickedOnHeader: boolean =
      clickedElementClassList.contains('solution-entry__header__name')
   || clickedElementClassList.contains('solution-entry__header__icon');
```

Comment (robpal94) [reply src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]: hmm, in your solution the `||` are before the `const` keyword but I think I got your idea.

**Comment by Vyperus** [comment]
This should be done by drag and drop.

**Pull request closed without merging**